packages/flutter_tools/lib/src/build_system/depfile.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../base/file_system.dart';
+import '../base/platform.dart';
+import '../globals.dart';
+
+/// A class for representing depfile formats.
+class Depfile {
+  /// Create a [Depfile] from a list of [input] files and [output] files.
+  const Depfile(this.inputs, this.outputs);
+
+  /// Parse the depfile contents from [file].
+  ///
+  /// If the syntax is invalid, returns an empty [Depfile].
+  factory Depfile.parse(File file) {
```

Comment:
This looks untested.

Reply:
It's tested indirectly via integration tests, but I've added some unit tests to cover it